invisible_cities/cities/detsim_test.py

```diff
@@ -0,0 +1,103 @@
+import os
+import shutil
+import numpy  as np
+import pandas as pd
+import tables as tb
+
+from .. cities.detsim  import detsim
+from .. core.configure import configure
+
+from invisible_cities.core  import system_of_units as units
+
+
+def test_detsim_contains_all_tables(ICDATADIR, config_tmpdir):
+
+    PATH_IN  = os.path.join(ICDATADIR    , "nexus_new_kr83m_fast.newformat.sim.h5")
+    PATH_OUT = os.path.join(config_tmpdir, "detsim_test.h5")
+    conf = configure('detsim $ICTDIR/invisible_cities/config/detsim.conf'.split())
+
+    buffer_length     = 800 * units.mus
+    pmt_width  = 100 * units.ns
+    sipm_width = 1   * units.mus
+
+    buffer_params = dict()
+    buffer_params["length"]      = buffer_length
+    buffer_params["pmt_width"]   = pmt_width
+    buffer_params["sipm_width"]  = sipm_width
+    buffer_params["max_time"]    =  10 * units.ms
+    buffer_params["pre_trigger"] = 100 * units.mus
+    buffer_params["trigger_thr"] = 0
+
+    conf.update(dict(files_in      = PATH_IN,
+                     file_out      = PATH_OUT,
+                     run_number    = 0,
+                     event_range   = (0, 1),
+                     buffer_params = buffer_params))
+    result = detsim(**conf)
+
+    with tb.open_file(PATH_OUT, mode="r") as h5out:
+        assert hasattr(h5out.root,                  'MC')
+        assert hasattr(h5out.root,                 'Run')
+        assert hasattr(h5out.root,               'pmtrd')
+        assert hasattr(h5out.root,              'sipmrd')
+        assert hasattr(h5out.root, 'Filters/active_hits')
+
+        pmtrd  = h5out.root.pmtrd
+        sipmrd = h5out.root.sipmrd
+
+        assert pmtrd .shape == (1,   12, int(buffer_length/pmt_width))
+        assert sipmrd.shape == (1, 1792, int(buffer_length/sipm_width))
+
+
+def test_detsim_filter_active_hits(ICDATADIR, config_tmpdir):
+
+    PATH_IN  = os.path.join(ICDATADIR    , "nexus_new_kr83m_fast.newformat.sim.h5")
+    PATH_OUT = os.path.join(config_tmpdir, "detsim_test.h5")
+    conf = configure('detsim $ICTDIR/invisible_cities/config/detsim.conf'.split())
+
+    # modify labels for event_id = 0 in test file
+    # all its hits labels are set to "NO_ACTIVE"
+    modified_testfile = os.path.join(config_tmpdir, "modified_testfile.h5")
```

Comment:
the files are less than 100KB, is probably ok to include them in the standard test suite